open-cluster-management/backlog#12184 Add pools

cadawson opened the pull request.

cadawson committed:
open-cluster-management/backlog#12184 Add pools

chrisahl commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+[#claim_prerequisites]
+== Prerequisites
+
+You must have the following prerequisites before claiiming a cluster from a cluster pool:
```
Typo. And also wondering what content to put in here for the ???  @showeimer ?

showeimer replied:
Not sure there is really a concrete pre-req, but a cluster can only be claimed from the cluster pool once it's available; the UI indicates the clusters as available.  So if there are 0 available clusters to claim, the Claim cluster button will be removed.

showeimer commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+
+*Note:* When a cluster that was claimed from the cluster pool is no longer needed and is destroyed, the resources are deleted. The cluster does not return to the cluster pool.
+
+*Required access*: Cluster administrator
```
I think this will be `admin`

cadawson replied:
In the docs, we are using either "cluster administrator" or "cluster-admin". In most cases, the "cluster administrator" format is used. I would prefer to leave these as they are, for consistency.

showeimer replied:
So a `cluster-admin` and an `admin` are two distinct roles in OpenShift

cadawson replied:
Understand that now. Thanks.

showeimer commented on clusters/cluster_pool_destroy.adoc:
```diff
@@ -0,0 +1,24 @@
+[#destroying-a-cluster-pool]
+= Destroying a cluster pool
+
+If you created a cluster pool and determine that you no longer need it, you can destroy the cluster pool. When you destroy a cluster pool, you release the resources of the hibernating clusters that are in that cluster pool. 
```
I think I understand the `release the resources` part here; but should it explicitly state that the clusters in the cluster pool will be destroyed?

showeimer replied:
Also I wonder if we should log a "known issue" that when you destroy a cluster pool, it will disappear from the UI even though all of the unclaimed clusters haven't finished destroying yet, so you lose ability to see them in UI.

cadawson replied:
Is this a "problem" that is likely to be fixed, or is it working as designed? A known issue is normally something that is a problem that is likely to be fixed or change soon. This case sounds like it might be just how it works, so I added a sentence in the procedure to point that out. See if that addresses the concern, or if I am missing the point.

chrisahl replied:
I think this is working as designed.  Would changing the description help?

If you created a cluster pool and determine that you no longer need it, you can destroy the cluster pool. When you destroy a cluster pool, all the unclaimed hibernating clusters will be destroyed and their resources will be released.

showeimer replied:
@chrisahl it kind of is working as designed, but imo the UX is a bit odd that the cluster pool is disappeared from the screen before all the unclaimed clusters are destroyed, I have an enhancement opened for Hive:  https://issues.redhat.com/browse/HIVE-1557

cadawson replied:
Does the sentence that I added to the file cover this one: `The clusters in the destroyed cluster pool are removed from the console while they are destroyed.`?

cadawson replied:
Sorry @chrisahl . I thought the issue there was based on the clusters disappearing from the UI. I tweaked the description a little to align more with your comment.

showeimer commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,68 @@
+
+The procedure for creating a cluster pool is similar to the procedure for creating a cluster. Clusters in a cluster pool are not created for immediate use.
+
+*Required access*: Cluster administrator
```
I think this also should be `admin`

cadawson replied:
In the docs, we are using either "cluster administrator" or "cluster-admin". In most cases, the "cluster administrator" format is used. I would prefer to leave these as they are, for consistency.

showeimer replied:
See above.

cadawson replied:
Got it now. Updated both of those.

showeimer commented on clusters/cluster_pool_rel_img_update.adoc:
```diff
@@ -0,0 +1,29 @@
+
+When the clusters in your cluster pool remain in hibernation for some time, the {ocp} release image of the clusters might become backlevel. If this happens, you can upgrade the version of the release image of the clusters that are in your cluster pool.  
+
+*Required access*: Cluster administrator
```
`edit`

showeimer commented on clusters/cluster_pool_rel_img_update.adoc:
```diff
@@ -0,0 +1,29 @@
+*Tip:* You can update the release image for multiple cluster pools with one action by selecting the box for each of the the cluster pools and using the _Actions_ menu to update the release image for the selected cluster pools.
+
+// CD: Are you limited to 4.8 images, or later for this, or can you use 4.7 versions?
+// CD: Why would you select an earlier version of the release image? 
```
They can select any version technically.  It's probably not likely they will select an earlier version, but they can.

chrisahl replied:
Some users need to use a specific OCP release image that is not necessarily the latest OCP image

showeimer commented on clusters/cluster_pool_rel_img_update.adoc:
```diff
@@ -0,0 +1,29 @@
+
+Complete the following steps to update the {ocp-short} release image for the clusters in your cluster pool:
+
+*Note:* This procedure does not update clusters from the cluster pool that are already claimed. It updates the release images for all of the remaining hibernating clusters that are created from this cluster pool after you complete this procedure. 
```
Not sure about the second sentence, what will end up happening is for example.  If you had two unclaimed clusters in the pool that are version 4.6.1, and you update the release image for the pool, those clusters will not changes to the new version.  Only after the release image is updated, any new clusters created after that will be the new version.

chrisahl replied:
So this is my misunderstanding of what happens when the image it updated for the cluster pool.  

New proposal:
*Note:* This procedure does not update clusters from the cluster pool that are already claimed and it does not update clusters in the pool that are hibernating.   Only new clusters that are created for the cluster pool will use the new release image after you complete this procedure.

cadawson replied:
I misunderstood this one, but it is updated now.

chrisahl commented on clusters/cluster_pool_rel_img_update.adoc:
```diff
@@ -0,0 +1,29 @@
+
+*Tip:* You can update the release image for multiple cluster pools with one action by selecting the box for each of the the cluster pools and using the _Actions_ menu to update the release image for the selected cluster pools.
+
+// CD: Are you limited to 4.8 images, or later for this, or can you use 4.7 versions?
```
Cluster pools will work with 4.7

chrisahl reviewed (changes requested):
Looks pretty good.  Only had a few comments.

cadawson committed:
open-cluster-management/backlog#12184 Rev

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson commented:
Addressed your comments in this PR @chrisahl and @showeimer . I know you both might be out today, so don't stress about this today. but please take a look when you have a chance. Thanks!

cadawson committed:
open-cluster-management/backlog#12184 fix link

chrisahl reviewed (changes requested):
I added a comment about destroy cluster pool

ldpliu commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,68 @@
+
+. Enter the namespace where you want to create your cluster pool. You can select an existing namespace from the list, or type the name of a new one to create one. The cluster pool does not have to be in the same namespace as the clusters.
+
+. Optional: Select a cluster set name if you want the RBAC roles for your cluster pool to share the role assignments of an existing cluster set. 
```
Note: If you are non-cluster-admin, you must specify the clusterset, if not, the create request will be forbiden. if there is no clusterset can be selected, please connect cluster-admin,he need create a clusterset and give clusterset admin permission to you.

ldpliu commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,68 @@
+
+. Enter a name for your cluster pool.
+
+. Enter the namespace where you want to create your cluster pool. You can select an existing namespace from the list, or type the name of a new one to create one. The cluster pool does not have to be in the same namespace as the clusters.
```
add note here: 
when you create a clusterpool in a clusterset, the whole namespace admin permission will be shared to clusterset admin users. the namespace view permission will be shared to clusterset view users.

ldpliu replied:
As we still discuss the clusterpool namespace permission in https://github.com/open-cluster-management/backlog/issues/13734#issuecomment-869577198
So we may need to wait the solution done, then we can decide what should be add here.
@cadawson you could also merge it first, if there is something changed, I will submit another pr to trace it.

ldpliu replied:
we will keep the current change, and we should add more notice in this part. So the desired description should be:
` 
When you create a cluster pool in a cluster set, the `namespace admin` permission is applied to all of the users with `clusterset admin` permissions for the namespace where you add the cluster pool. Similarly, the `namespace view` permission is applied to the users with `clusterset view` permissions. 

So as cluster admin, you should create one namespace for each team to create clusterpool.
`
@qiujian16 please also review the description here.

cadawson replied:
I think this is resolved, @ldpliu , is that correct? @qiujian16 , are you ok with the updates?

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson commented:
New updates in @showeimer and @chrisahl . Hopefully, that resolves the requests. Thanks!

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson committed:
open-cluster-management/backlog#12184 rev

cadawson commented:
Hi @ldpliu . I added the notes that you requested to clarify the permissions limitations. Please see if that addresses your concerns. Thank you!

dockerymick commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+[#claiming-clusters-from-cluster-pools]
+= Claiming clusters from cluster pools
+
+ClusterClaim resources are used to check out clusters from cluster pools. After a cluster is checked out from a cluster pool, the checked out cluster is brought to a ready state. The cluster pool automatically creates a new hibernated cluster in the cluster pool to replace the claimed one.
```
I recommend ClusterClaim be monospaced

dockerymick commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+
+*Note:* When a cluster that was claimed from the cluster pool is no longer needed and is destroyed, the resources are deleted. The cluster does not return to the cluster pool.
+
+*Required access*: admin
```
change admin to Administrator

cadawson replied:
Admin is the name of the role. Are you still using administrator? Just want to be consistent. Changed to administrator.

dockerymick commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+
+You must have the following available before claiming a cluster from a cluster pool:
+
+* A cluster pool with available clusters to claim. See xref:../clusters/cluster_pool_manage.adoc#creating_a_cluster_pool.adoc[Creating a cluster pool] for information about how to create a cluster pool. If there are no available clusters to claim, the *Claim cluster* button is not displayed.
```
Since there is only one prerequisite, change Prerequisites to Prerequisite. 

I also recommend removing the bullet point:

You must have a cluster pool with available clusters to claim.

dockerymick commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+[#claim_cluster]
+== Claim the cluster from the cluster pool
+
+To checkout a cluster from the cluster pool, resume the cluster and claim it as one of your managed clusters. 
```
Add "Complete the following steps to claim a cluster:"

dockerymick commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+        
+. From the navigation menu, click *Infrastructure* > *Clusters*.
+
+. Select the *Cluster pools* tab.
```
Cluster pool should be italicized

dockerymick commented on clusters/cluster_claim.adoc:
```diff
@@ -0,0 +1,32 @@
+
+. Select *Claim*.
+
+. The claimed cluster comes out of hibernation and appears in the *Managed Cluster* tab with a status of `Resuming`.  After the cluster reaches a status of  `Ready`, the cluster is available for use.
```
Remove the period prior to "The". This works better as a concluding statement rather than another task.

cadawson replied:
Good catch. It was intended to be the concluding sentence.

dockerymick commented on clusters/cluster_pool_destroy.adoc:
```diff
@@ -0,0 +1,24 @@
+
+. From the navigation menu, click *Infrastructure* > *Clusters*.
+
+. Select the *Cluster pools* tab.
```
Italicize Cluster pool

dockerymick commented on clusters/cluster_pool_destroy.adoc:
```diff
@@ -0,0 +1,24 @@
+
+. Select *Destroy* to remove the cluster pool. The clusters in the destroyed cluster pool are removed from the console while they are destroyed.
+
+*Tip:* You can destroy multiple cluster pools with one action by selecting the box for each of the the cluster pools and using the _Actions_ menu to destroy the selected cluster pools.
```
I would move the tip to line 17 

Then I recommend adding a concluding sentence that reads: Your cluster pool is deleted.

cadawson replied:
Trying to keep the procedure clean without adding the tip there. Added the conclusion.

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+
+Cluster pools provide rapid and cost-effective access to configured {ocp} clusters on-demand and at scale. Cluster pools maintain a configurable and scalable number of {ocp-short} clusters on Amazon Web Services, Google Cloud Platform, or Microsoft Azure in a hibernating state that can be claimed when they are needed. They are especially useful when providing or replacing cluster environments for ...
+
+ClusterClaim resources are used to checkout clusters from cluster pools. After a cluster is checked out from a cluster pool, the checked out cluster is resumed and brought to a ready state. The cluster pool will automatically create a new hibernated cluster, to replace the claimed one, in the cluster pool.
```
I recommend monospacing ClusterClaim

dockerymick replied:
Remove 'will' 

The cluster pool automatically creates a new hibernated cluster to replace the claimed one in the cluster pool.

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+[#pool_prerequisites]
+=== Prerequisites
+
+You must have the following prerequisites before creating a cluster pool:
```
I would move the Prerequisites before Creating a cluster pool. I would also add an inline link for prerequisites

Change admin to Administrator

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+
+You must have the following prerequisites before creating a cluster pool:
+
+* A deployed {product-title} hub cluster
```
Add a period at the end of this statement

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+
+. From the navigation menu, click *Infrastructure* > *Clusters*.
+
+. Select the *Cluster pools* tab to view cluster pools that you can access.
```
Italicize Cluster pool

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+
+. Optional: Select a cluster set name if you want the RBAC roles for your cluster pool to share the role assignments of an existing cluster set. 
++
+*Note:* If you do not have `cluster admin` permissions, you must select a cluster set. The request to create a cluster set is rejected as forbidden if you do not include the cluster set name in this situation. If no cluster sets are available for you to select, contact your cluster administrator to create a cluster set and give you `clusterset admin` permissions to it.
```
i recommend bolding optional, but i understand why you might choose not too considering the bold note that comes right after


The request to create a cluster set is rejected and forbidden if you do not include the cluster set name in this situation.

cadawson replied:
Clarified this  a little.

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+* xref:../clusters/create_google.adoc#creating-a-cluster-on-google-cloud-platform[Creating a cluster on Google Cloud Platform]
+* xref:../clusters/create_azure.adoc#creating-a-cluster-on-microsoft-azure[Creating a cluster on Microsoft Azure]
+
+10. After all of the information is entered, select *Create* to create the cluster pool. After at least one cluster is available in the pool, a *Claim cluster* button appears.
```
This step should be on line 64 and also remove `10`, there should only be a period prior to the task.

cadawson replied:
I agree with the 10. That was a miss. 
For the step, the references are tied to the information that they are entering in the second-to-last step. I think it is in the right order.

dockerymick commented on clusters/cluster_pool_manage.adoc:
```diff
@@ -0,0 +1,72 @@
+* <<scaling-a-clusterpool,Scaling a cluster pool>>
+* <<updating-the-clusterpool-release-image,Updating the cluster pool release image>>
+* <<deleting-a-clusterpool,Deleting a cluster pool>>
+
```
This inline links are not working. All of the links after Creating a cluster pool doesn't work. Were you attempting to link to the other files or are you planning to add more sections?

cadawson replied:
Was trying to link the files, but I will have to look at this.

dockerymick commented on clusters/cluster_pool_rel_img_update.adoc:
```diff
@@ -0,0 +1,26 @@
+
+. From the navigation menu, click *Infrastructure* > *Clusters*.
+
+. Select the *Cluster pools* tab.
```
Cluster pool should be italicized

dockerymick commented on clusters/cluster_pool_rel_img_update.adoc:
```diff
@@ -0,0 +1,26 @@
+      
+. Select *Update* to apply the selected release image.
+
+*Tip:* You can update the release image for multiple cluster pools with one action by selecting the box for each of the the cluster pools and using the _Actions_ menu to update the release image for the selected cluster pools.
```
Move this tip to line 17 and conclude the doc with the following statement:

The cluster pool release image is updated.

cadawson replied:
Added the conclusion. Trying to keep the procedure clean, so not moving the tip.

dockerymick commented on clusters/cluster_pool_scale.adoc:
```diff
@@ -0,0 +1,20 @@
+
+. From the navigation menu, click *Infrastructure* > *Clusters*.
+
+. Select the *Cluster pools* tab.
```
Cluster pools should be italicized

dockerymick commented on clusters/cluster_pool_scale.adoc:
```diff
@@ -0,0 +1,20 @@
+
+. Increase or decrease the pool size by changing the value of the pool size.
+
+. Select *Scale* to commit the change.
```
Add a conclusion statement similar to the following sentence: Your cluster pools are scaled.

dockerymick reviewed (commented):
I completed a review of the PR yesterday, please let me know if you do not see my comments @cadawson

qiujian16 commented:
/approve

qiujian16 commented:
/lgtm

cadawson reviewed (commented):
Thank you for the review. Always appreciated!

cadawson committed:
open-cluster-management/backlog#12184 peer

cadawson committed:
open-cluster-management/backlog#12184 Add wht new

cadawson committed:
Merge branch '2.3_stage' into dawson1218423
